### PR TITLE
Order execution is getting there

### PR DESCRIPTION
**Pull request by TaylorLewis**
It's not there yet, but here's what I've thrown together so far.
Let me know if I'm doing this right.

At this point it seems I'd next need to add some Database functions (?):
- Insert wallet transaction
- Delete stock transaction
- Delete wallet transaction

_(Just ignore the engine.go changes.)_

### CODE REVIEW

File: backend/internal/service/matching/execute.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+import "day-trading-app/backend/internal/service/models"
+
+var s serviceImpl
+
+func ExecuteOrders(stockTxCommitQueue []models.StockMatch) {
+	for _, tx := range stockTxCommitQueue {
+		if isParent(tx) {
+			// Update stock transaction as completed
+			s.db.UpdateStockOrderStatus(tx.Order.UserName, tx.Order.StockTxID, tx.Order.OrderStatus)
```
Comment (matheus-ds) [inline backend/internal/service/matching/execute.go:11]: that is not going to work. You are creating a new var for the serviceImpl so it wont have db initialized. It's an empty struct.

File: backend/internal/service/models/stocks.go [hunk 2]
```diff
@@ -32,7 +32,8 @@ type StockPrice struct {
 }
 
 type StockMatch struct {
-	Order      StockTransaction `json:"order" bson:"order"`
-	QuantityTx int              `json:"quantityTx" bson:"quantity_tx"`
-	PriceTx    int              `json:"priceTx" bson:"price_tx"`
+	Order       StockTransaction `json:"order" bson:"order"`
+	QuantityTx  int              `json:"quantityTx" bson:"quantity_tx"`
+	PriceTx     int              `json:"priceTx" bson:"price_tx"`
+	CostTotalTx int              `json:"costTotalTx" bson:"cost_total_tx"`
```
Comment (matheus-ds) [inline backend/internal/service/models/stocks.go:38]: json values need to be `snake_case`, not `camelCase`. that's what the TestDoc expects

File: backend/internal/service/matching/execute.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+	} else {
+		// Error: spelling probably
+	}
+}
+
+func isParent(tx models.StockMatch) bool {
+	return (tx.Order.OrderStatus != "IN_PROGRESS") && (tx.PriceTx == 0)
+}
+
+type Database interface {
```
Comment (matheus-ds) [inline backend/internal/service/matching/execute.go:87]: this is not how you import the DB handler into another package. This wont work and needs to be deleted

File: backend/internal/service/models/stocks.go (hunk 2 above)
Comment (TaylorLewis) [reply backend/internal/service/models/stocks.go:38]: I was just copying the format of the structs above it. Would this mean all the structs in models/stocks.go must be changed?

File: backend/internal/service/matching/execute.go (hunk 3 above)
Comment (jluover9000) [reply backend/internal/service/matching/execute.go]: Actually give me a list of things you want to manipulate in the database, like add something, update something, delete something. I can create that function for you. I feel like we should have only the store layer talking to the db cuz later if I want to make changes to our data model, we’d have to adjust engine/execution and store layer which is not efficient.

**Pull request closed without merging**